### DISCUSSION TITLE
Is there a preferred method for allowing X or Y dependency for a formulae?

### DISCUSSION THREAD

**Discussion by kylekorth** [Homebrew/discussions#4037] [Writing Formulae/Casks]
For example in [cdktf.rb on line 21](https://github.com/Homebrew/homebrew-core/blob/2830bfe9da2cb4d357e5255b2bb4ba024dffce4c/Formula/cdktf.rb#L21) exists a requirement for terraform. There is another formulae ([tfenv](https://github.com/Homebrew/homebrew-core/blob/2830bfe9da2cb4d357e5255b2bb4ba024dffce4c/Formula/tfenv.rb#L24)) which symlinks the terraform binary on a system so you can switch back and forth between versions. 

I wanted to add some logic to the dependency on terraform to allow tfenv as a substitute. I started writing out a simple if statement because I couldn't find an OR classification in the depends_on selector but it felt messy (not to mention depends_on is a class method so it cannot be nested inside of an if statement).

```
if Formula.installed.include?("tfenv")
  depends_on "tfenv"
else
  depends_on "terraform"
end
```
vs something like:
`depends_on :optional => ["terraform", "tfenv"]`

Any advice on moving forward here? 
Thanks!

**Comment by chenrui333** [comment]
I tried to persuade the team on [this PR](https://github.com/Homebrew/homebrew-core/pull/111627), I think we should mark all terraform dependencies as optional, I will raise a new one for discussion.

**Reply by kylekorth** [reply]
Yeah it looks like removing it as a dependency was a precedent set with Terragrunt:
https://github.com/Homebrew/homebrew-core/pull/106362/files 

tfsec doesn't even list it as a dependency:
https://github.com/Homebrew/homebrew-core/blob/master/Formula/tfsec.rb
